core/chains/evm/logpoller/orm.go

```diff
@@ -323,462 +348,405 @@ func (o *DbORM) validateLogs(logs []Log) error {
 	if err != nil {
 		return nil, err
 	}
 
-	nestedQuery := nestedBlockNumberQuery(confs)
+	nestedQuery := nestedBlockNumberQuery(confs, ubig.New(o.chainID))
 	query := fmt.Sprintf(`
 		SELECT * FROM   evm.logs
-		WHERE   evm_chain_id = :evm_chain_id
-		AND     address = :address
-		AND     event_sig = :sigA
-		AND 	block_number BETWEEN :start_block AND :end_block
+		WHERE   evm_chain_id = $1
+		AND     address = $2
+		AND     event_sig = $3
+		AND 	block_number BETWEEN $5 AND $6
 		AND 	block_number <= %s		
 		EXCEPT
 		SELECT     a.* FROM       evm.logs AS a
 		INNER JOIN evm.logs B
 		ON         a.evm_chain_id = b.evm_chain_id
 		AND        a.address = b.address
-		AND        a.topics[:topic_index] = b.topics[:topic_index]
-		AND        a.event_sig = :sigA
-		AND        b.event_sig = :sigB
-	    AND 	   b.block_number BETWEEN :start_block AND :end_block
+		AND        a.topics[$7] = b.topics[$7]
+		AND        a.event_sig = $3
+		AND        b.event_sig = $4
+	    AND 	   b.block_number BETWEEN $5 AND $6
 		AND		   b.block_number <= %s
 		ORDER BY block_number,log_index ASC`, nestedQuery, nestedQuery)
+
 	var logs []Log
-	if err := o.q.WithOpts(qopts...).SelectNamed(&logs, query, args); err != nil {
+	ctx, cancel := context.WithTimeout(ctx, defaultTimeout)
+	defer cancel()
+	if err := o.db.SelectContext(ctx, &logs, query, ubig.New(o.chainID), address, sigA.Bytes(), sigB.Bytes(), startBlock, endBlock, topicIndex); err != nil {
 		return nil, err
 	}
 	return logs, nil
 }
 
-func nestedBlockNumberQuery(confs Confirmations) string {
+func nestedBlockNumberQuery(confs Confirmations, chainID *ubig.Big) string {
 	if confs == Finalized {
-		return `
+		return fmt.Sprintf(`
 				(SELECT finalized_block_number 
 				FROM evm.log_poller_blocks 
-				WHERE evm_chain_id = :evm_chain_id 
-				ORDER BY block_number DESC LIMIT 1) `
+				WHERE evm_chain_id = %v 
+				ORDER BY block_number DESC LIMIT 1) `, chainID)
 	}
 	// Intentionally wrap with greatest() function and don't return negative block numbers when :confs > :block_number
 	// It doesn't impact logic of the outer query, because block numbers are never less or equal to 0 (guarded by log_poller_blocks_block_number_check)
-	return `
-			(SELECT greatest(block_number - :confs, 0) 
+	return fmt.Sprintf(`
+			(SELECT greatest(block_number - %d, 0) 
 			FROM evm.log_poller_blocks 	
-			WHERE evm_chain_id = :evm_chain_id 
-			ORDER BY block_number DESC LIMIT 1) `
+			WHERE evm_chain_id = %v 
+			ORDER BY block_number DESC LIMIT 1) `, confs, chainID)
+
+}
+
+func UseTopicIndex(index int) (int, error) {
+	// Only topicIndex 1 through 3 is valid. 0 is the event sig and only 4 total topics are allowed
+	if !(index == 1 || index == 2 || index == 3) {
+		return 0, fmt.Errorf("invalid index for topic: %d", index)
+	}
+	// Add 1 since postgresql arrays are 1-indexed.
+	return index + 1, nil
+}
 
+type bytesProducer interface {
+	Bytes() []byte
+}
+
+func concatBytes[T bytesProducer](byteSlice []T) pq.ByteaArray {
```

Comment:
Not sure if this would affect typing but couldn't we just return `[][]byte` here? Should help us avoid the extra pq import

Reply:
Awesome!